### PR TITLE
CPOL-130 If the engine does not have the policy, we can safely delete it

### PR DESCRIPTION
**Pull request by pilhuhn**
Signed-off-by: Heiko W. Rupp <hrupp@redhat.com>

### CODE REVIEW

File: src/main/java/com/redhat/cloud/custompolicies/app/rest/PolicyCrudService.java [hunk 1]
```diff
@@ -338,8 +339,16 @@ public Response deletePolicy(@PathParam("id") UUID policyId) {
...
-      builder = Response.ok(policy);
+      try {
+        engine.deleteTrigger(policy.id, user.getAccount());
+        builder = Response.ok(policy);
+      } catch (NotFoundException nfe) {
+        // Engine does not have it - we can delete anyway
+        builder = Response.ok(policy);
+      } catch (Exception e) {
+        e.printStackTrace();  // TODO: Customise this generated block
+        builder = Response.serverError().entity(new Msg(e.getMessage()));
```
Comment (zsadeh) [inline src/main/java/com/redhat/cloud/custompolicies/app/rest/PolicyCrudService.java]: which error code is it going to produce? is it server error 500?

Comment (pilhuhn) [reply src/main/java/com/redhat/cloud/custompolicies/app/rest/PolicyCrudService.java]: yes

File: src/main/java/com/redhat/cloud/custompolicies/app/rest/PolicyCrudService.java [hunk 2]
```diff
@@ -338,8 +339,16 @@ public Response deletePolicy(@PathParam("id") UUID policyId) {
...
-      engine.deleteTrigger(policy.id, user.getAccount());
-      builder = Response.ok(policy);
+      try {
+        engine.deleteTrigger(policy.id, user.getAccount());
+        builder = Response.ok(policy);
+      } catch (NotFoundException nfe) {
+        // Engine does not have it - we can delete anyway
+        builder = Response.ok(policy);
+      } catch (Exception e) {
+        e.printStackTrace();  // TODO: Customise this generated block
```
Comment (burmanm) [inline src/main/java/com/redhat/cloud/custompolicies/app/rest/PolicyCrudService.java]: Probably shouldn't be part of the commit?
